packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js

```diff
@@ -75,106 +88,102 @@ class CloudFormation {
 
-            return Promise.all(nestedStackEventPromises)
-              .then((nestedstackEvents) => {
-                newEvents = nestedstackEvents.reduce((combinedEventList, nestedstackEventList) =>
-                  combinedEventList.concat(nestedstackEventList), newEvents);
-                // Just get the new events to display to the user
+          if (!nestedStackQueue.length) {
+            // empty queue
+            nestedStackQueue = nestedStacks;
+          } else {
+            // maintain the order of the queue and remove missing stack
+            nestedStackQueue = _.intersection(nestedStackQueue, nestedStacks);
 
-                newEvents = _.differenceBy(newEvents, allShownEvents, 'EventId');
-                newEvents = _.uniqBy(newEvents, 'EventId');
-                showEvents(newEvents);
-                allShownEvents = allShownEvents.concat(newEvents);
-              });
+            // add newly created stack to the end of the queue
+            const newStacks = _.difference(nestedStacks, nestedStackQueue);
+            nestedStackQueue = [...nestedStackQueue, ...newStacks];
           }
+
+          for (
+            let i = 0;
+            i < Math.min(CFN_MAX_CONCURRENT_REQUEST, nestedStackQueue.length);
+            i += 1
+          ) {
+            nestedStackEventPromises.push(self.getNestedStackEvents(nestedStackQueue[0]));
+            if (nestedStackQueue.length > 1) {
+              nestedStackQueue.shift();
+            }
+          }
+
+          return Promise.all(nestedStackEventPromises).then((nestedstackEvents) => {
+            newEvents = nestedstackEvents.reduce(
+              (combinedEventList, nestedstackEventList) =>
+                combinedEventList.concat(nestedstackEventList),
+              newEvents,
+            );
+            // Just get the new events to display to the user
+
+            newEvents = _.differenceBy(newEvents, allShownEvents, 'EventId');
+            newEvents = _.uniqBy(newEvents, 'EventId');
+            showEvents(newEvents);
+            allShownEvents = allShownEvents.concat(newEvents);
+          });
         }
       });
-    }, 10000); // Poll for any events every 10 seconds
+    }, CFN_POLL_TIME);
   }
 
   getNestedStackEvents(stackName) {
     const self = this;
-    return this.cfn.describeStackEvents({ StackName: stackName }).promise()
+    return this.cfn
+      .describeStackEvents({ StackName: stackName })
+      .promise()
       .then((data) => {
         let nestedstackEvents = data.StackEvents;
-        nestedstackEvents = nestedstackEvents.filter(event =>
-          self.eventStartTime < new Date(event.Timestamp));
+        nestedstackEvents = nestedstackEvents
+          .filter(event => self.eventStartTime < new Date(event.Timestamp));
 
         return nestedstackEvents;
-      });
+      })
+      .catch(() =>
+        // ignore. We dont fail if we can't get the nested stack status
```

Comment:
I think we should have a check for the CFN errors. There might be other errors as well which will go uncaught and hard to debug in a production environment.

Reply:
OK